### ISSUE TITLE
BUG: Cannot Login - Unless I Reload the Browser or App

### ISSUE THREAD

**Issue by Syirrus** [RocketChat/Rocket.Chat#3761]
Your Rocket.Chat version: (make sure you are running the latest)

I don't know if anyone is experiencing this issue, but for some reason when I try to login to Rocket.Chat, my browser or phone gets stuck on the login loading screen. 

When I refresh the browser or reload (force close and run) the app I can log into my account account again. 

I have tried different accounts which doesn't seem to make a difference. 

![cannot_login](https://cloud.githubusercontent.com/assets/138841/16793214/823e8a3c-48cf-11e6-86ec-da6e47871c1d.jpg)

Syirrus

**Comment by Syirrus** [comment]
I have narrowed down to the problem and I even recreated it on https://rocket.chat

Basically if you login and log out right away and log back in again all works fine. 
If you login click on a channel or anything other than /home and then log out and then log back in the screen hangs on the loading screen. I've tried this in Chrome, Chrome Incognito mode (no browser extensions loaded) and on the mobile cordova app with the same results.

I've tried logging in with a FB account and just a plain email with the same results.

This is very problematic if people receive a push notification and go to log in on their mobiles and cannot. 

<img width="1920" alt="screen shot 2016-07-13 at 8 26 26 am" src="https://cloud.githubusercontent.com/assets/138841/16793785/d047bbb4-48d3-11e6-9b0d-f60ce91cfb58.png">

**Comment by waley1975** [comment]
Yes,I have the same problem.

**Comment by Syirrus** [comment]
I'm surprised this has flown under the radar for so long :(.

**Comment by geekgonecrazy** [comment]
As per comments in #4009 re-opening this issue.  It still happening at least for some people.. We need to figure out why.

**Comment by bhbarbosa** [comment]
Can confirm, it happens also on both Firefox and IE, normal and incognito modes.

**Comment by ryannealmes** [comment]
Happens for us on Chrome :/

**Comment by CONFORM-IT** [comment]
Same thing here.

**Comment by rodrigok** [comment]
Is this happening in our demo server or in your installations?

**Comment by stealthybox** [comment]
**+1**
We see this bug only after logging out and trying to log back in.
It occurs for both local users and LDAP users.

Refreshing the app is a temporary workaround.

( Running the official 0.39.0 docker image from the hub )

**Comment by LostInAzure** [comment]
+1 (0.39.0)
Syirrus wrote a fix in another issue but how do I patch that fix into an existing deployment without having to rebuild the entire application?

**Comment by stealthybox** [comment]
@LostInAzure is this patch in a branch somewhere?
Could you link the issue? Thanks :hamburger:

**Comment by flaviogrossi** [comment]
The problem seems to be that while rendering the main template the `subsReady` helper [here](https://github.com/RocketChat/Rocket.Chat/blob/develop/packages/rocketchat-ui-master/master/main.coffee#L153l) is false and for some reason the rendering doesn't update when it becomes true.

Trying to understand why...

**Comment by rodrigok** [comment]
@flaviogrossi Please see the PR #4541

**Comment by flaviogrossi** [comment]
@rodrigok thanks, i can confirm this fixes the issue for me

**Comment by Syirrus** [comment]
This doesn't fix the issue for me. You can duplicate it on https://demo.rocket.chat I logged in as a FB user and clicked on a channel and then logged out. When you try to log immediately back in, your stuck on the load screen.

**Comment by rodrigok** [comment]
@Syirrus There was a PR for the fix, it wasn't merged and deployed to demo.

**Comment by Syirrus** [comment]
@rodrigok I see, my mistake. I did try the PR change and it does address the problem. Thank you very much. :)
